File: JAVA/Ruster exercise/com/Ruster/Person.java


/**
 *
 * @author JUAN
 */
package com.Ruster;

public abstract class Person {
    private static final int DefaultDaysWorked= 30;
    
    private String name;
    private int identification;
    private double valueOfTheDay;
    private int daysWorked;
    private double salary;
    private double subTransport;
    
    public Person(String name, int identification, double valueOfTheDay, Integer daysWorked){
        this.name= name;
        this.identification= identification;
        this.daysWorked= (daysWorked!=null) ?daysWorked : DefaultDaysWorked;
        this.valueOfTheDay= valueOfTheDay;
        this.salary= this.daysWorked * this.valueOfTheDay;
        this.subTransport= (this.salary<2000000) ?114000 :0;
        
            }
    public String getName(){
        return this.name;
        
    }
    
    public int getIdentification(){
        return this.identification;
    }
    
    public int getdaysWorked(){
        return this.daysWorked;
    }
    
    public double getValueOfTheDay(){
        return this.valueOfTheDay;
    }
    
    public double getSalary(){
        return this.salary;
    }
    
    public double getSubTransport(){
        return this.subTransport;
    }
    public abstract String getDescription();
    public abstract double getNetSalary();
    


}

  

File: PHP/Ruster exercise/Ruster/Person.php
<?php
abstract class Person {
    private const DefaultDaysWorked =26;

    private string $name;
    private int $identification;
    private float $valueOftheDay;
    private int $daysWorked;
    private float $salary;
    private float $subTransport;

    public function __construct(string $name, int $identification, float $valueOftheDay, ?int $daysWorked=null) {
        $this->name = $name;
        $this->identification = $identification;
        $this->daysWorked = $daysWorked ?? self::DefaultDaysWorked;
        $this->valueOftheDay = $valueOftheDay;
        $this->salary = $this->daysWorked * $this->valueOftheDay;
        $this->subTransport = ($this->salary < 2000000) ? 114000 : 0;
    }

    public function getName(): string {
        return $this->name;
    }

    public function getIdentification(): int {
        return $this->identification;
    }

    public function getDaysWorked(): int {
        return $this->daysWorked;
    }

    public function getValueOftheDay(): float {
        return $this->valueOftheDay;
    }

    public function getSalary(): float {
        return $this->salary;
    }

    public function getSubTransport(): float {
        return $this->subTransport;
    }

    public abstract function getDescription(): string;
    public abstract function getNetSalary(): float;
}
